### PR TITLE
Updates to support node 4 and above.

### PR DESCRIPTION
**Pull request by mattaylor**

### CODE REVIEW

**Comment by jadonk** [comment]
Please work against 0.5.x branch. I'll move it to master as soon as I add a suitable kernel version check.

File: package.json [hunk 1]
```diff
@@ -26,7 +26,7 @@
     "node": ">= 0.10.1"
   },
   "dependencies": {
-    "socket.io": "3.1.2",
+    "socket.io": "latest",
```
Comment (jadonk) [inline package.json]: I'd like to avoid using "latest" due to surprise breakage.

**Comment by jadonk** [comment]
@mattaylor Any chance you can rebase? If iic-bus is better, I might switch. I think @RobertCNelson was making the same recommendation. Have you actually tried it?

**Comment by RobertCNelson** [comment]
@jadonk just noticed, "i2c" is alive again:

https://github.com/kelly/node-i2c/commits/master

Regards,

**Comment by RobertCNelson** [comment]
@jadonk if you pull in i2c 0.2.3, remember to tag bonscript as 0.6.0 as nodejs v0.12.x needs to bumped to v4.x

**Comment by jadonk** [comment]
OK, anything pulling in i2c 0.2.3 would be a beta version only. Attempts to find solutions to pull in multiple versions of a module have turned up to be dead ends.

**Comment by jadonk** [comment]
Here's what I did: https://github.com/jadonk/bonescript/commit/ad374711ba1cf95cb95ffcfa883a7efad5ddac2a

**Pull request closed without merging**